runtime/cudaq/qis/qubit_qis.h

```diff
@@ -990,4 +573,38 @@ std::vector<T> slice_vector(std::vector<T> &original, std::size_t start,
   return ret;
 }
 
-} // namespace cudaq
\ No newline at end of file
+} // namespace cudaq
+
+#define __qop__ __attribute__((annotate("user_custom_quantum_operation")))
+
+/// Register a new custom unitary operation providing a unique name,
+/// the number of target qubits, the number of rotation parameters (can be 0),
+/// and the unitary matrix as a 1D row-major `std::vector<complex>`
+/// representation following a MSB qubit ordering.
+#define CUDAQ_REGISTER_OPERATION(NAME, NUMT, NUMP, ...)                        \
```

Comment:
If interpret the count of targets by accepting unitary as a 2-D matrix, and then interpret any additional argument as a parameter to get count of parameters, can we drop the `NUMT` and `NUMP` arguments?

Reply:
For the current implementation, we need the integers at compile time (for the tuple processing), so I'm not sure we can use the unitary data which is provided as a std::vector. Maybe I'm wrong though.